resources/notes.py

```diff
@@ -26,4 +26,12 @@ class Note(Resource):
     def delete(self, ticket_id, id):
         ticket = TicketModel.find(ticket_id)
         ticket.notes.delete(NotesModel.find(id))
+
         return {"message": "Note deleted"}
+
+    @pm_level_required
+    def patch(self, ticket_id, id):
+        ticket = TicketModel.find(ticket_id)
+        note = ticket.notes.find(id)
+
+        return note.update(schema=NotesSchema, payload=request.json, id=id).json()
```

Comment:
This looks good. The `update` method looks like its an instance method here, but it is actually calling the class method which I find a bit confusing. Also the id parameter seems a bit weird. If we are calling update on note then update should know that the object being updated is note, and shouldn't need the id. 

**Option A.**
I would like to see this slightly improved. Either call update on the class and make it clear which id is being used.
```python
NoteModel.update(schema=NotesSchema, payload=request.json, id=note.id
```

However, I don't quite like that, because for one: We already have the note object. And this class method will hit the database again to find the note object. When I created these class method they had made sense at the time, but they're not static. They can be evolved and improved.

**Option B.**
I think I would rather see an instance method called `update` that updates an instance.

```python
note.update(schema=NotesSchema, payload=request.json)
```

**Extra Credit**
It would also be nice to not have to pass in the schema argument and be able to either infer that or ask the object for the schema.

```python
note.update(request.json)
```

Reply:
That is all possible, but wouldn't mean messing with the .update method in the BaseModel? I'm happy to do it, but I wanted to make sure that is what you are envisioning.

Reply:
Let me think about this some more. I'm very unnerved by the fact that you can call a class level method from an instance of the object. Which means there is no separation between class and instance methods in python. It looks like they all share the same namespace. That does not make me a happy programmer.

`note.update(schema=NotesSchema, payload=request.json, id=id)` <- This looks like an instance method, but it's a class method.

Reply:
I put in PR #329 to address this issue. Review it. Let me know.